presto-main/src/main/java/com/facebook/presto/type/DoubleOperators.java

```diff
@@ -217,7 +217,12 @@ public static long castToTinyint(@SqlType(StandardTypes.DOUBLE) double value)
     @SqlType(StandardTypes.BIGINT)
     public static long castToLong(@SqlType(StandardTypes.DOUBLE) double value)
     {
-        return (long) MathFunctions.round(value);
+        try {
+            return toIntExact((long) MathFunctions.round(value));
```

Comment:
`toIntExact` converts the 64bit long value to a 32bit int value, but `BIGINT` is a 64 bit value.